ls1intum/Artemis
# `Development`: Improve the setup documentation

### Checklist
#### General




### Motivation and Context


The documentation for [*Setting Up* Artemis](https://docs.artemis.ase.in.tum.de/dev/setup/) was really confusing for me when I first joined the project.
After some time of getting to know the different moving parts, I saw that the structure of the documentation wasn't consistent.
It seems like a lot of things are categorized beneath the **Server Setup** which makes it confusing.
I guess that the transformation from pure Markdown to using ReadTheDocs (see #1462) led to this, at least from my perspective, "inconsistent" structure.

### Description

Mainly there is now [one big *Setup* page](https://artemis-platform--5648.org.readthedocs.build/en/5648/dev/setup/) with sections for all of the subsystems necessary to operate Artemis.
Some of the subsystems' documentation might be in separate documents due to legacy and decoupling reasons.
Another option would have been to strictly separate all of the documents but it seemed better to me to have one big page with a nice navigation on the left side.

**UPDATE:** There is already a follow-up issue #5657 for this PR which should split the setup guide into contributor/developer vs administration/deployment concerns.

There are a lot of cosmetics that got changed like shortening lines, correcting typos, and naming errors I found while going through.

### Steps for Testing

Prerequisites:
- 1 Developer

1. Go to the [documentation preview on readthedocs](https://artemis-platform--5648.org.readthedocs.build/en/5648/dev/setup/)
2. Compare and check the structure of both versions and if it looks correct
3. Look for anything weird looking from a formatting point of view

### Review Progress





#### Code Review
#### Manual Tests
